vignettes/task_plans.Rmd

```diff
@@ -0,0 +1,255 @@
+```{r plan}
+task_plan_1 <- create_task_plan(
+  task_config$id,
+  list(step1, step2, step3),
+  final_steps='report',
+  ind_dir='states/log')
+```
+
+You'll need the `task_plan` in that form for two things: to create a remake .yml makefile and to run `loop_tasks`. First, the makefile:
+
+```{r makefile, results='markup'}
+create_task_makefile(
+  task_plan=task_plan_1,
+  makefile='task_plan_1.yml',
+  include=c(),
+  packages=c('ggplot2'),
+  sources=c('script.R'),
+  file_extensions=c('ind'),
+  ind_complete=TRUE)
+```
+
+And here are the contents of task_plan_1.yml:
+
+```{r read_makefile, results='markup', comment='', echo=FALSE}
+cat(readr::read_lines('task_plan_1.yml'), sep='\n')
+```
+
+Note that at the end of each state's task list is a final task, e.g., `states/log/WI.ind`, that we didn't directly specify in the task_plan. These recipes for the `complete` step were generated in `create_task_plan` because we allowed the default argument `add_complete=TRUE`.
+
+Next, use the `task_plan` and the `task_makefile` in `loop_tasks`:
+
+```{r delete1, echo=FALSE}
+unlink('.remake')
+unlink('build')
+```
+
+```{r loop1a}
+loop_tasks(
+  task_plan=task_plan_1,
+  task_makefile='task_plan_1.yml')
+```
+
+## Forcing rebuilds
+
+### Forcing rebuild for one task-step
+
+If we try to `scmake` any of the targets in the task makefile, `remake` correctly recognizes that they're up to date already, and the target is not rebuilt. Instead, `AZ_plot.png` is already `OK`:
+```{r try_rebuild_one}
+scmake('AZ_plot.png', 'task_plan_1.yml')
+```
+
+One way to force the rebuild of a single target is to use `scdel`, like this:
+```{r delete_one}
+scdel('AZ_plot.png', 'task_plan_1.yml')
+```
+
+Now when we `scmake` that target again, `AZ_plot.png` gets rebuilt:
+```{r rebuild_one}
+scmake('AZ_plot.png', 'task_plan_1.yml')
+```
+
+### Forcing rebuild for all task-steps
+
+If we run the same `loop_tasks` command a second time, nothing will get rebuilt. Instead, all items say `OK`:
+```{r try_rebuild_all}
+loop_tasks(
+  task_plan=task_plan_1,
+  task_makefile='task_plan_1.yml')
+```
+
+One way to force a complete rebuild (not generally a good use of time, but sometimes necessary) is to use `scdel` with the `target_names` argument either missing or set to `NULL`. We'll do this several times in this vignette for demonstration purposes.
+```{r delete_all}
+scdel(remake_file='task_plan_1.yml')
+```
+
+Now when we run the entire loop, everything gets rebuilt:
+```{r rebuild_all}
+loop_tasks(
+  task_plan=task_plan_1,
+  task_makefile='task_plan_1.yml')
+```
+
+## Developing a successful loop
+
+Not analyses will go smoothly the first time you try. You can use `loop_tasks` to run just a few task-steps at a time at first, debugging before you launch a full batch.
+
+Before we can demonstrate, we'll need to delete all output files and remake status information again:
+```{r delete_all2}
+scdel(remake_file='task_plan_1.yml')
```

Comment:
`scdel` seems to return a boolean for each task (three `TRUE`s here). Should that be included in the `scdel` `return` doc?

Reply:
nah, we could just return invisible(). will change.